### DISCUSSION TITLE
Is it possible to remove the background of a theme?

### DISCUSSION THREAD

**Discussion by hazel-ocean** [helix-editor/helix#4092] [Q&A]
I'm a big fan of the themes helix provides, but sometimes my terminal's background does not match the theme I would like to choose. Is it possible to disable the background color of a theme to allow my terminal theme's background to take precedence?

Something similar to what base_16_transparent does to the background of helix, but applying that to pre-existing themes.

I mean, I could copy all the themes helix provides and make my own variants of them, but it would be nice to have a way to just unset the background on all themes with a toggle or some global override.

**Comment by CptPotato** [comment]
Currently this requires editing the theme file. You can set the background to transparent by commenting/removing the corresponding line from the theme you want to use:

```toml
"ui.background" = "..."
```

**Comment by rafaelrcamargo** [comment]
That's usually done by replacing a theme prop, what I typically do is to set the `ui.background` field to some property that's not defined down at palette in the theme file, something like that:"

```toml
"ui.background" = {bg = "bg_transparent"}
...
``` 
> I use it in my dotfiles, I have a gruvbox material theme [here!](https://github.com/rafaelrcamargo/dotfiles/blob/master/helix/themes/grv_material.toml#L53)

**Comment by hazel-ocean** [comment]
Can I make a suggestion for a feature that would allow someone to say, "ignore painting the ui's background color regardless of what the theme says"?

For me, personally, it is very distracting to see the surrounding terminal a different color than Helix's UI background.

jokingly: I get that having one data point doesn't make for a very compelling reason to add a feature, soo, how about I raise you two data points? https://github.com/helix-editor/helix/discussions/4236

I can see this not being desirable if the implementation causes problems if Helix were to offer UIs other than a terminal UI, and that the only reasonable course would be to ignore options that don't make sense outside of the context of running inside a terminal.

Still, it would be nice to have a config file option like the following:

```toml
theme = "gruvbox"

[terminal.ui]
ignore_ui_background = true
```

Or to provide a way to override themes at the global level:

```toml
theme = "gruvbox"

[theme.override]
"ui.background" = { bg = None }
```

**Comment by pathwave** [comment]
I think having a way to override parts of the theme in the config file would be nice.

**Comment by C0DK** [comment]
isn't this already sorta fixed?

**Reply by tasnuva1** [reply]
How? I'm still didn't finding it.

**Reply by 2gn** [reply]
> How? I'm still didn't finding it.

This worked for me btw

https://github.com/helix-editor/helix/discussions/10223#discussioncomment-9032722

<img width="1303" height="846" alt="image" src="https://github.com/user-attachments/assets/2d0c5e67-3d76-4692-9ea6-a52f5a92badf" />
